lib/coverband/adapters/base.rb

```diff
@@ -124,8 +130,15 @@ def merge_expanded_data(new_expanded, old_expanded)
         }
       end
 
+      # TODO: This should only be 2 cases get our dup / not dups aligned
       def array_add(latest, original)
-        latest.map.with_index { |v, i| (v && original[i]) ? v + original[i] : nil }
+        if Coverband.configuration.use_oneshot_lines_coverage
+          latest.map!.with_index { |v, i| (v + original[i] >= 1 ? 1 : 0) if v && original[i] }
```

Comment:
yeah I thought of this, but I think it ends up a bit more complex...

```
if Coverband.configuration.use_oneshot_lines_coverage
          latest.map!.with_index { |v, i| (v + original[i] >= 1 ? 1 : 0) if v && original[i] }
        elsif Coverband.configuration.simulate_oneshot_lines_coverage
          latest.map.with_index { |v, i| (v + original[i] >= 1 ? 1 : 0) if v && original[i] }
```

becomes

```
if (Coverband.configuration.use_oneshot_lines_coverage || Coverband.configuration.simulate_oneshot_lines_coverage)
  if latest.frozen?
       latest.map.with_index { |v, i| (v + original[i] >= 1 ? 1 : 0) if v && original[i] }
  else
       latest.map!.with_index { |v, i| (v + original[i] >= 1 ? 1 : 0) if v && original[i] }
```

I would like to update the code so we have a single strategy all the way through, but it will be a bit of a bigger effort and I am hoping to combine that with dropping support for older Ruby etc and do a full cleanup pass.

Thinking we always get the hash from `Coverage` and try to filter it early to a isolated object, then use all in memory reductions beyond the delta class.

Reply:
Gotcha. By one strategy, do you mean always use oneshot?

Reply:
In this case, I am more thinking of a strategy for how we process Coverage data... like having rules where we try to never dup and always filter in-memory vs into copies...